src/java/org/apache/cassandra/index/sai/SSTableIndex.java

```diff
@@ -73,13 +80,35 @@ public SSTableIndex(SSTableContext sstableContext, IndexComponents.ForRead perIn
     {
         assert perIndexComponents.context().getValidator() != null;
         this.perIndexComponents = perIndexComponents;
-        this.searchableIndex = perIndexComponents.version().onDiskFormat().newSearchableIndex(sstableContext, perIndexComponents);
+        this.searchableIndex = createSearchableIndex(sstableContext, perIndexComponents);
 
         this.sstableContext = sstableContext.sharedCopy(); // this line must not be before any code that may throw
         this.indexContext = perIndexComponents.context();
         this.sstable = sstableContext.sstable;
     }
 
+    private static SearchableIndex createSearchableIndex(SSTableContext sstableContext, IndexComponents.ForRead perIndexComponents)
```

Comment:
should we also disable per-sstable stuff as well? in `onDiskFormat#newPrimaryKeyMapFactory`

Reply:
I've been wondering, but I wasn't 100% certain that it wasn't used as part of index writing and I don't think that's the part that we worry about, so I wanted to keep it targeted/de-risk as much as possible. I can take some time to check tomorrow though.